ui/src/router/components/form/components/router_config/RoutesPanel.js

```diff
@@ -34,27 +19,18 @@ export const RoutesPanel = ({
   const onDeleteRoute = (idx) => () => {
     routes.splice(idx, 1);
     onChange("routes")(routes);
-    idx < defaultRouteIndex && setDefaultRouteIndex((idx) => idx - 1);
   };
 
-  const defaultRouteIdUpdated = routes[defaultRouteIndex].id;
-
-  useEffect(() => {
-    onChange("default_route_id")(defaultRouteIdUpdated);
-  }, [defaultRouteIdUpdated, onChange]);
-
   return (
     <Panel title="Routes">
       <EuiFlexGroup direction="column" gutterSize="s">
         {routes.map((route, idx) => (
           <EuiFlexItem key={`route-${idx}`}>
             <RouteCard
               route={route}
-              isDefault={idx === defaultRouteIndex}
               endpointOptions={endpoints}
               onChange={onChange(`routes.${idx}`)}
-              onSelect={() => setDefaultRouteIndex(idx)}
-              onDelete={onDeleteRoute(idx)}
+              onDelete={routes.length > 1 ? onDeleteRoute(idx) : null}
```

Comment:
Previously, we weren't allowing the default route to be deleted. Now that this concept is removed, we only try to ensure there is always at least one route.